File: solutions/fortran/bob/1/bob.f90
module bob
   implicit none
   character(*), parameter :: lowercaseLetters = "abcdefghijklmnopqrstuvwxyz"
   character(*), parameter :: uppercaseLetters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
   character(*), parameter :: letters = lowercaseLetters//uppercaseLetters
contains
   pure function hey(statement)
      character(:), allocatable :: hey
      character(*), intent(in) :: statement
      character(:), allocatable :: trimmedStatement
      logical :: isYelled, isQuestion, isEmpty, hasLetters

      trimmedStatement = trim(adjustl(statement))
      isEmpty = trimmedStatement == ""
      hasLetters = scan(trimmedStatement, letters) > 0
      isYelled = scan(trimmedStatement, lowercaseLetters) == 0 .and. hasLetters
      isQuestion = trimmedStatement(len(trimmedStatement):len(trimmedStatement)) == '?'

      if (isEmpty) then
         hey = "Fine. Be that way!"
      else if (isYelled .and. isQuestion) then
         hey = "Calm down, I know what I'm doing!"
      else if (isYelled) then
         hey = "Whoa, chill out!"
      else if (isQuestion) then
         hey = "Sure."
      else
         hey = "Whatever."
      end if
   end function hey
end module bob

File: solutions/clojure/bob/1/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn response-for [s]
  (let [s (str/trim s)
        question? (str/ends-with? s "?")
        yell? (and (re-matches #".*\p{L}.*" s)
                   (= s (str/upper-case s)))
        silent? (str/blank? s)
        any-char? ()]
    (cond
      silent? "Fine. Be that way!"
      (and question? yell?) "Calm down, I know what I'm doing!"
      question? "Sure."
      yell? "Whoa, chill out!"
      :else "Whatever.")))
